### DISCUSSION TITLE
RSM Helper Plugin

### DISCUSSION THREAD

**Discussion by dark22** [Rarisma/Rarismas-Server-Manager#4] [General]
Would it be possible to provide the source code of the TShock plugin used for console commands to work? Thanks for the work!

**Comment by Rarisma** [comment]
I actually lost it,
I was working on adding quick commands for Terraria and realised that I accidentally deleted the plugin's source,
Since terraria support isn't finished yet I'm going to remake it since it was fairly simple, when its done, Ill post a link here Ill probably put it in the server files directory

The plugin is functionally useless for anything but SSM since for some reason I can't interact with the window like for minecraft, 

SSMHelper basically
- Checks for SSM.txt (Might be a different name)
- Subscribes to changes for SSM.txt
- When it changes it reads the file and executes the command as server

Ideally I should be able to interact with a terraria server via StandardInput but it just ignores any input sent.

**Reply by dark22** [reply]
I had already opened the plugin compiled with ILSpy and I already knew that its function was only to read the txt file and execute the commands from there. But I appreciate your attention and time when explaining about it. Terraria has a problem with console redirection and although TShock has some changes in relation to that, it also doesn't seem to work well. There are other ways to do this, such as Redis, or client and server via TCP, create a plugin to receive this via RESTAPI already present in TShock, among others.

There are interactions about TShock console redirection at: https://github.com/Pryaxis/TerrariaAPI-Server/pull/185
Example of how to pass commands through TCP: https://github.com/LaurensDeV/Remote-TShock-Console

They are just ideas, I am not proposing changes, since your solution already works as desired.

Thank you very much for your attention, I wish you success with your project!

**Reply by Rarisma** [reply]
Those are some interesting solutions, I might look into using the TCP plugin if it's easy to use

If that pull request gets merged then I'll try and use the regular redirection since it would make the CLIServer smaller

**Reply by Rarisma** [reply]
The SSMHelper plugin works however there are some instances where it straight up crashes terraria

**Reply by Rarisma** [reply]
@dark22  I remade the plugin and added it [here](https://github.com/Rarisma/Simple-Server-Manager/tree/main/ServerFiles/Terraria/SSMHelper%20Source%20Code), I changed some things to prevent spam but I just decompiled it mostly and copy and pasted the code, This version has a a cool down of 1 second after running a command.
